airflow/models/connection.py

```diff
@@ -144,6 +145,7 @@ def parse_from_uri(self, uri):
         self.port = uri_parts.port
         if uri_parts.query:
             self.extra = json.dumps(dict(parse_qsl(uri_parts.query, keep_blank_values=True)))
+        self.uri_without_query = uri_parts.scheme + "://" + uri_parts.netloc + uri_parts.path
```

Comment:
what if the URI equals to 'yarn' or 'local'?

Reply:
oh yep, Just updated the commit.